src/I18n/TranslatorRegistry.php

```diff
@@ -227,54 +299,26 @@ public function useFallback(bool $enable = true): void
     }
 
     /**
-     * Returns a new translator instance for the given name and locale
-     * based of conventions.
-     *
-     * @param string $name The translation package name.
-     * @param string $locale The locale to create the translator for.
-     * @return \Aura\Intl\TranslatorInterface|\Closure
-     */
-    protected function _fallbackLoader(string $name, string $locale)
-    {
-        return $this->_loaders[$this->_fallbackLoader]($name, $locale);
-    }
-
-    /**
-     * Returns a function that can be used as a loader for the registerLoaderMethod
+     * Set fallback domain for package.
      *
-     * @return \Closure
+     * @param string $name The name of the package.
+     * @param \Cake\I18n\Package $package Package instance
+     * @return \Cake\I18n\Package
      */
-    protected function _partialLoader(): Closure
+    public function setFallbackPackage(string $name, Package $package): Package
     {
-        return function ($name, $locale) {
-            return $this->_fallbackLoader($name, $locale);
-        };
-    }
-
-    /**
-     * Registers a new package by passing the register loaded function for the
-     * package name.
-     *
-     * @param string $name The name of the translator package
-     * @param string $locale The locale that should be built the package for
-     * @return \Aura\Intl\TranslatorInterface A translator object.
-     */
-    protected function _getFromLoader(string $name, string $locale): TranslatorInterface
-    {
-        $loader = $this->_loaders[$name]($name, $locale);
-        $package = $loader;
-
-        if (!is_callable($loader)) {
-            $loader = function () use ($package) {
-                return $package;
-            };
+        if ($package->getFallback()) {
+            return $package;
```

Comment:
Why would the setter return the package being set?

Reply:
The fallback is meant to be set on the package object itself. So if it already has a fallback set nothing has to be done and same object is returned back without any modification.

Reply:
Sure. I just mean it's strange that the setter is returning anything.